### PR TITLE
Ashwalkers start with Settler benefits (taming/riding/fishing)

### PR DESCRIPTION
**Pull request by Metekillot**
## About The Pull Request

This pull request gives the settler trait to ashwalkers, since they're native denizens of lavaland. The effect is that they are better at taming, riding, and fishing. It has no effect on their movespeed, hunger, or height, as they get the trait instead of the full quirk.
## Why It's Good For The Game

Ashwalkers should be good at these things inherently.
## Changelog

Ashwalkers now have the benefits of the settler trait.
:cl:
balance: Ashwalkers are now as good at taming, fishing, and riding as other planet-borne settlers!
code: Ashwalkers not being able to use the shuttle consoles has been made a trait of their brain, instead of inherent to their species (for cases such as someone being forcibly made into an ashwalker, or brain transplants, etc)
/:cl:

### CODE REVIEW

File: code/modules/antagonists/ashwalker/ashwalker.dm [hunk 1]
```diff
@@ -28,6 +28,7 @@
 	. = ..()
 	RegisterSignal(owner.current, COMSIG_MOB_EXAMINATE, PROC_REF(on_examinate))
 	owner.teach_crafting_recipe(/datum/crafting_recipe/skeleton_key)
+	owner.add_traits(list(TRAIT_SETTLER), ROLE_ASHWALKER)
```
Comment (MrMelbert) [inline code/modules/antagonists/ashwalker/ashwalker.dm:31]: Why is this `add_traits` if there's only 1 trait

Comment (Metekillot) [reply code/modules/antagonists/ashwalker/ashwalker.dm:31]: oop

File: code/modules/antagonists/ashwalker/ashwalker.dm [hunk 2]
```diff
@@ -28,6 +28,7 @@
 	. = ..()
 	RegisterSignal(owner.current, COMSIG_MOB_EXAMINATE, PROC_REF(on_examinate))
 	owner.teach_crafting_recipe(/datum/crafting_recipe/skeleton_key)
+	ADD_TRAIT(owner, TRAIT_SETTLER, ROLE_ASHWALKER)
```
Comment (MrMelbert) [inline code/modules/antagonists/ashwalker/ashwalker.dm:31]: Also untested code alert, you're putting this trait on the mind of the mob, not the mob itself. 

Also also this should really be in `apply_innate_effects` / `remove_innate_effects`.

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/ashwalker/ashwalker.dm (hunk 2 above)
Comment (Metekillot) [reply code/modules/antagonists/ashwalker/ashwalker.dm:31]: Well now that you mention it, I see that it's their species that makes it so ashwalkers can't use the shuttle consoles. Shouldn't that be something determined either by their mind, or perhaps their brain? And the same with settler stuff.

Comment (Metekillot) [reply code/modules/antagonists/ashwalker/ashwalker.dm:31]: Or as you say, inside apply_innate_effects

Comment (MrMelbert) [reply code/modules/antagonists/ashwalker/ashwalker.dm:31]: Could be a brain trait, it'd fit

**Comment by carlarctg** [comment]
they should be small

**Comment by necromanceranne** [comment]
The trait has a lot more plugged into it than just survivalist stuff. It also lets them move faster while wearing heavy equipment. That part is balanced with the expectation that they're already slow by default.

It would be a good idea to separate the traits out from a single quirk and give them traits for the parts you want to keep, or just shrug and give them the whole quirk. Not like that's a bad thing :U

**Comment by Jacquerel** [comment]
> It would be a good idea to separate out the benefits from a single trait into multiple different traits, and give them traits for the parts you want to keep.

I would prefer this solution, specific traits are generally safer than generalist ones which are going to accumulate extra behaviour over time.

**Comment by san7890** [comment]
a really bright way to do this is to rework the settler trait into a defined list, so

```dm
//old
#define TRAIT_SETTLER "old trait"

//new
#define TRAIT_BETTER_FISHER "a"
#define TRAIT_BETTER_TAMER "b"
#define TRAIT_BETTER_RIDER "c"
#define SETTLER_TRAITS list(TRAIT_BETTER_FISHER, TRAIT_BETTER_TAMER, TRAIT_BETTER_RIDER) // might have to be in a different file? i forget how smart the linters are

// now we can simply replace every time we were doing ADD_TRAIT(src, TRAIT_SETTLER, INNATE_TRAIT) with something like the following
add_traits(SETTLER_TRAITS, INNATE_TRAIT)
```

So, if someone wants to expand SETTLER_TRAITS to give people with the Settler quirk more stuff, it is far easier to update that one define instead of having to update the quirk and whatever else might use it. Everything in the code will need to be updated to access each individual component of the trait, but hopefully this idea helps simplify the infrastructure rather than copypaste the same list a few times.

**Comment by Metekillot** [comment]
That looks good to me. I'll need to open a new PR because the scope on this one doesn't encompass a refactor of quirk code and it'd be confusing to anyone searching PRs/commits down the line.

**Pull request closed without merging**